k8s/cronjob-reboot/cm.yaml

```
@@ -0,0 +1,43 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: restart-deployment-configmap
+  namespace:  palworld
+data:
+  restart-deployment.sh:  |
+    #!/bin/bash
+    cont=$(kubectl -n palworld get pods -o=jsonpath='{.items[?(@.metadata.labels.app=="palworld-server")].metadata.name}')
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Saving_Server_Data..."
+    kubectl exec -n palworld -i pod/$cont rcon-cli save
+    sleep 30
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Backing_up_Server_Data..."
+    kubectl exec -n palworld -i pod/$cont backup
+    sleep 30
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Backup_Complete!_Rebooting_in_30_seconds..."
+    sleep 15
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_15_seconds..."
+    sleep 5
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_10_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_9_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_8_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_7_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_6_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_5_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_4_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_3_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_2_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Broadcast Rebooting_in_1_seconds..."
+    sleep 1
+    kubectl exec -n palworld -i pod/$cont rcon-cli "Shutdown 1"
+    sleep 30
+    kubectl -n palworld rollout restart deployment/palworld-server
```

can you make a for loop out of that? also set the namespace via env or as a variable on top like cont, it is way less error prone

---

I will sit down and look closer at this for improvement

---

Following should do the job:

```suggestion
  restart-deployment.sh:  |
    #!/bin/bash
    cont=$(kubectl -n palworld get pods -o=jsonpath='{.items[?(@.metadata.labels.app=="palworld-server")].metadata.name}')

    function exec_rcon_cmd() {
      kubectl exec -n palworld -i pod/$cont rcon-cli "$1"
    }

    exec_rcon_cmd "Broadcast Saving_Server_Data..."
    exec_rcon_cmd save
    sleep 30
    exec_rcon_cmd "Broadcast Backing_up_Server_Data..."
    exec_rcon_cmd backup
    sleep 30

    for i in {30..1}; do
      exec_rcon_cmd "Broadcast Rebooting_in_$i_seconds..."
      sleep 1
    done
    
    exec_rcon_cmd "Shutdown 1"
    sleep 30
    kubectl -n palworld rollout restart deployment/palworld-server
```

---

You could also move the foor loop around and have it like 

```
for i in {0..30}
do
  exec_rcon_cmd "Broadcast Rebooting_in_$((30-($i*5)))_seconds..."
  sleep 5
done
```